internal/server/iito.go

```diff
@@ -15,3 +23,84 @@ func (s *SOMASServer) runIITOEndOfTurn() error {
 	// TODO:- IITO team
 	return nil
 }
+
+func (s *SOMASServer) runGiftSession() error {
+	s.logf("start runGiftSession")
+	defer s.logf("finish runGiftSession")
+
+	giftRequestDict := s.getGiftRequests()
+	giftOffersDict, err := s.getGiftOffers(giftRequestDict)
+	if err != nil {
+		return err
+	}
+	giftHistoryDict, err := s.getGiftAcceptance(giftOffersDict)
+	if err != nil {
+		return err
+	}
+	err = s.distributeGiftHistory(giftHistoryDict)
+	if err != nil {
+		return err
```

Comment:
unwrapped errors make me feel uneasy but I trust your functions to return errors that are descriptive enough

(the errors have stack traces, so hopefully they're ok)

![image](https://user-images.githubusercontent.com/33488131/102896835-ed131f00-445e-11eb-932b-1864aa6106b5.png)

Reply:
Ehehe, yeah the plan was basically to add it in for now to have less work later but I can add another PR post MVP to add some more rigourous error handling